OpenTokMeet/app/src/main/java/com/opentok/android/meet/ChatRoomActivity.java

```diff
@@ -106,64 +139,104 @@ public void onCreate(Bundle savedInstanceState) {
             View cView = getLayoutInflater().inflate(R.layout.custom_title, null);
             actionBar.setCustomView(cView);
         }
-
-        mPreview = (ViewGroup) findViewById(R.id.publisherview);
-        mParticipantsView = (LinearLayout) findViewById(R.id.gallery);
-        mLastParticipantView = (ViewGroup) findViewById(R.id.mainsubscriberView);
-        mLoadingSub = (ProgressBar) findViewById(R.id.loadingSpinner);
-
-        Uri url = getIntent().getData();
-        serverURL = getResources().getString(R.string.serverURL);
-
-        if (url == null) {
-            mRoomName = getIntent().getStringExtra(ARG_ROOM_ID);
-            mUsername = getIntent().getStringExtra(ARG_USERNAME_ID);
+        /* parse out intent data */
+        final Intent intent = getIntent();
+        if (null == intent.getData()) {
+            mRoomName       = intent.getStringExtra(INTENT_ROOM_NAME);
+            mUsername       = intent.getStringExtra(INTENT_USER_NAME);
+            mApiKey         = intent.getStringExtra(INTENT_APIKEY);
+            mSessionId      = intent.getStringExtra(INTENT_SESSION_ID);
+            mSessionToken   = intent.getStringExtra(INTENT_SESSION_TOKEN);
+            mCapturerRes    = RESOLUTION_TBL.get(intent.getStringExtra(INTENT_CAP_RESOLUTION));
+            mCapturerFps    = FRAMERATE_TBL.get(intent.getStringExtra(INTENT_CAP_FPS));
+            /* update UI to reflect information received */
+            ((TextView)findViewById(R.id.title)).setText(mRoomName);
+            /* setup room */
+            mRoom = new Room(this, mSessionId, mSessionToken, mApiKey, mUsername);
+            mRoom.setPreviewView(mPreview);
+            mRoom.setParticipantsViewContainer(mParticipantsView, mLastParticipantView, null);
+            mRoom.connect();
         } else {
-            if (url.getScheme().equals("otmeet")) {
-                mRoomName = url.getHost();
-            } else {
-                mRoomName = url.getPathSegments().get(0);
-            }
-        }
-
-        TextView title = (TextView) findViewById(R.id.title);
-        title.setText(mRoomName);
-
-        statsInfo.add("CPU info stats are not available");
-        statsInfo.add("Memory info stats are not available");
-        statsInfo.add("Battery info stats are not available");
-
-        String resolution =  getIntent().getStringExtra(PUB_CAPTURER_RESOLUTION);
-        mCapturerResolutionPub = getPubCapturerResolution(resolution);
-
-        String framerate =  getIntent().getStringExtra(PUB_CAPTURER_FPS);
-        mCapturerFpsPub = getPubCapturerFrameRate(framerate);
+            /* fetch api key/session Id/aession Token from server */
```

Comment:
That was needed because the chatroom activity can be launched via a url link.. That could be moved to the homescreen activity, but it wasn't in the past so I didn't change that flow. I can move it if you would like